doc/developers/advanced_installation.rst

```diff
@@ -12,137 +15,219 @@ Installing nightly builds
 
 The continuous integration servers of the scikit-learn project build, test
 and upload wheel packages for the most recent Python version on a nightly
-basis to help users test bleeding edge features or bug fixes::
+basis.
+
+Installing a nightly build is the quickest way to:
+
+- try a new feature that will be shipped in the next release (that is, a
+  feature from a pull-request that was recently merged to the master branch);
+
+- check whether a bug you encountered has been fixed since the last release.
+
+::
 
   pip install --pre -f https://sklearn-nightly.scdn8.secure.raxcdn.com scikit-learn
 
 
 .. _install_bleeding_edge:
 
 Building from source
-=====================
+====================
+
+Building from source is required to work on a contribution (bug fix, new
+feature, code or documentation improvement).
+
+.. _git_repo:
+
+#. Use `Git <https://git-scm.com/>`_ to check out the latest source from the
+   `scikit-learn repository <https://github.com/scikit-learn/scikit-learn>`_ on
+   Github.::
+
+        git clone git://github.com/scikit-learn/scikit-learn.git
+        cd scikit-learn
+
+   If you plan on submitting a pull-request, you should clone from your fork
+   instead.
+
+#. Install a compiler with OpenMP_ support for your platform. See intructions
+   for :ref:`compiler_windows`, :ref:`compiler_macos`, :ref:`compiler_linux`
+   and :ref:`compiler_freebsd`.
```

Comment:
Maybe we could add a note that for most linux distros this will just work out of the box?

And if we go for the new solution, maybe we can just say that openmp support isn't critical for most contributions?

Reply:
You still to install `build-essential` on debian / ubuntu for instance. gcc is not installed by default and first time contribution data scientist might never have installed any C/C++ compilers prior to contributing to scikit-learn at a scipy sprint for instance.